pytext/data/data_handler.py

```diff
@@ -641,27 +641,35 @@ def _get_test_iter(
             is_train=False,
         )
 
-    def get_predict_iter(self, data: List[Dict[str, Any]]):
+    def get_predict_iter(
+        self, data: List[Dict[str, Any]], batch_size: Optional[int] = None
+    ):
         ds = self.gen_dataset(data, include_label_fields=False)
+        if batch_size is None:
+            batch_size = len(ds)
         it = BatchIterator(
             textdata.Iterator(
                 ds,
-                batch_size=len(ds),
+                batch_size=batch_size,
                 device="cuda:{}".format(torch.cuda.current_device())
                 if cuda_utils.CUDA_ENABLED
                 else "cpu",
                 sort=True,
                 repeat=False,
                 train=False,
                 sort_key=self.sort_key,
+                sort_within_batch=self.sort_within_batch,
             ),
             self._postprocess_batch,
             include_target=False,
             is_train=False,
         )
-        for input, _, context in it:
-            # only return the first batch since there is only one
-            return input, context
+        if batch_size is not None:
+            return it
```

Comment:
This breaks `Task.predict` and thus `pytext predict-py`, as a 2-tuple is assumed to be returned. Also the `if`-predicate is always true because `batch_size` gets defaulted to `len(ds)` above.